File: EjemplosJson/JavaScript/CRUDJavaScriptShowTotalInJsonEMPLOYEE/Controller/EmployeeController.js
const EmployeeCrud = require("../Model/EmployeeCrud");
const Employee = require("../Model/Employee");

class EmployeeController {

    constructor() {
        this.crud = new EmployeeCrud();
        this.payPerHour = 5.50; //PRICE FOR HOUR 
    }

    add(name, hoursWorked) {
        const id = this.crud.generateId();
        const totalPay = Number(hoursWorked) * this.payPerHour;
        const employee = new Employee(id, name, hoursWorked, totalPay);
        this.crud.addEmployee(employee);
        return employee;
    }

    getAll() {
        return this.crud.getAllEmployees();
    }


    edit(id, name, hoursWorked) {
        const totalPay = Number(hoursWorked) * this.payPerHour;
        const updatedEmployee = new Employee(id, name, hoursWorked, totalPay);
        const success = this.crud.editEmployee(id, updatedEmployee);
        return success; 
    }

    delete(id) {
        const success = this.crud.deleteEmployee(id);
        return success; 
    }
}

module.exports = EmployeeController;

File: EjemplosJson/CRUDPythonEmployee/Employee_controller.py
from EmployeeCrud_model import EmployeeCrud
from Employee_model import Employee

class EmployeeController:
    def __init__(self):
        self.crud = EmployeeCrud()
        self.pay_per_hour = 5.50  # PRICE FOR HOUR

    def add(self, name, hours_worked):
        id = self.crud.generate_id()
        total_pay = float(hours_worked) * self.pay_per_hour
        employee = Employee(id, name, hours_worked, total_pay)
        self.crud.add_employee(employee)
        return employee

    def get_all(self):
        return self.crud.get_all_employees()

    def edit(self, id, name, hours_worked):
        total_pay = float(hours_worked) * self.pay_per_hour
        updated_employee = Employee(id, name, hours_worked, total_pay)
        success = self.crud.edit_employee(id, updated_employee)
        return success

    def delete(self, id):
        success = self.crud.delete_employee(id)
        return success
